### PR TITLE
doma: Introduce XT_EXTERNAL_ANDROID_SOURCES

### PR DESCRIPTION
**Pull request by rshym**
Note: Based on not yet merged branch.

### CODE REVIEW

**Comment by rshym** [comment]
Directly depends on https://github.com/xen-troops/meta-xt-images/pull/197.
So, as result needs also
xen-troops/xt-distro#53
and 
https://github.com/xen-troops/meta-xt-prod-devel/pull/332

**Comment by rshym** [comment]
Tested. Works as expected: with XT_EXISTING_ANDROID_DIR pointing to folder with android sources, domu_image_android:do_fetch() was disabled, provided sources were used and only do_compile() and do_install() were executed.

**Comment by rshym** [comment]
Tested by build of prod-devel doma with variable not defined. Works as expected - sources are downloaded.

File: recipes-domu/domu-image-android/domu-image-android.bb [hunk 1]
```diff
@@ -1,76 +1,129 @@
...
+# Some details:
+# 'depth=1' reduces amount of data downloaded from server and is most significant
+#           reducer of fetch time
+# 'REPODIR' is used to download sources directly into required place,
+#           and to avoid copying of tens of gigabytes from ${DL_DIR}
+# 'directpath=1' is used to suppress creation of subfolders to ${REPODIR}.
+#           This is required for correct work of do_compile() with downloaded sources.
 
-S = "${WORKDIR}/repo"
+python __anonymous () {
```
Comment (iusyk) [inline recipes-domu/domu-image-android/domu-image-android.bb:27]: i think, the required parameters of url must be checked in the function 'anonymous', and if parameters are not defined or have the unexpected values - stop the processing.

**Review by iusyk — CHANGES_REQUESTED** [review]

**Comment by rshym** [comment]
This PR is based on #197 and can be considered as part of reworking of build of doma. So I will unite #197 and #200 to simplify review and maintenance.

**Pull request closed without merging**